### PR TITLE
Add sponsor logos to home page and footer

### LINKED ISSUE #269: Add sponsor-section
I open issue here; easier to make the changes here and then port it to the theme.

* Add to the footer and "somewhere" on the home page. A little subtle and not too big.
* See my mockup below. I want some placeholders to indicate that there is room for two more.

image

### PR DESCRIPTION
**Pull request by budparr**
This PR closes #269 

@bep I put the logos themselves on a white background for the home page and footer so that we only have to collect one logo per sponsor. I reworked the footer considerably. 

There is sample content in there, which I can remove. 

The sponsor boxes are triggered by the presence of  items in the `sponsors` key in the `_index.md` file. If there is an item without a logo, the copy "Your Logo Here" will appear. That, of course, can be blank, or anything else.

Lastly, as far as I can tell about Git Subtrees, we just pull this PR and then update the theme repo separately. Is that right, or even close to right?

### CODE REVIEW

File: themes/gohugoioTheme/layouts/index.html [hunk 1]
```diff
@@ -14,6 +14,9 @@
   <section class="w-100 ph4 ph5-ns pv4 pv6-ns mid-gray bg-white bb bt b--light-gray">
     {{- partial "home-page-sections/installation" . -}}
   </section>
+
+  {{ partial "home-page-sections/sponsors.html" (dict "cx" . ) }}
```
Comment (moorereason) [inline themes/gohugoioTheme/layouts/index.html]: Why pass a `dict` here?

Comment (budparr) [reply themes/gohugoioTheme/layouts/index.html]: So I can reuse the same partial in the footer and pass through some classes

**Comment by bep** [comment]
>Lastly, as far as I can tell about Git Subtrees, we just pull this PR and then update the theme repo separately. Is that right, or even close to right?

That is right. The syntax is a little bit clever, but I will fix it once we get this merged into this repo's master.

This looks good. Two minor comments.

1. Remove the logo etc. for the two boxes to the right so we get to see how it will end up like.
2. The "Your Logo Here" boxes need to be much more muted. Like mostly transparent, only indicating that "there is room for two more".

Other than that I think it looks great.

File: content/_index.md [hunk 2]
```diff
@@ -44,6 +44,19 @@ sections:
...
+  - name: "Forestry.io"
+    link: https://forestry.io/
+    logo: /images/sponsors/forestry-logotype.svg
+    copy: "Forestry is A full-featured CMS for your Hugo sites"
+  - name: "Netlify"
+    link: https://forestry.io/
+    logo: /images/sponsors/netlify-full-logo-light.svg
+    copy: "Write frontend code. Push it. We handle the rest."
+  - name: "Algolia"
+    link: https://forestry.io/
```
Comment (phrawzty) [inline content/_index.md]: Should probably be `https://algolia.com/` 😄

Comment (budparr) [reply content/_index.md]: That was just dummy content and it's been removed.

File: content/_index.md [hunk 3]
```diff
@@ -44,6 +44,19 @@ sections:
     color_classes: bg-primary-color-light black
     image: /images/home-page-templating-example.png
     copy: "Hugo's Go-based templating provides just the right amount of logic to build anything from the simple to complex. If you prefer Jade/Pug-like syntax, you can also use Amber, Ace, or any combination of the three."
+sponsors:
+  - name: "Forestry.io"
+    link: https://forestry.io/
+    logo: /images/sponsors/forestry-logotype.svg
+    copy: "Forestry is A full-featured CMS for your Hugo sites"
```
Comment (bep) [inline content/_index.md]: "is A" > "is a"

**Comment by bep** [comment]
Yes, this looks good.

**Comment by bep** [comment]
@scottgallant  what do you think about this? 

Preview at https://deploy-preview-270--gohugoio.netlify.com/

**Comment by scottgallant** [comment]
@bep  this is great! I'll ask our designer to make a PR with updated artwork and copy.

**Pull request merged**